python/mdvtools/llm/langchain_mdv.py

```
@@ -0,0 +1,468 @@
+from langchain_core.prompts import ChatPromptTemplate, MessagesPlaceholder
+from langchain_experimental.tools.python.tool import PythonAstREPLTool
+from langchain.agents import create_openai_functions_agent, AgentExecutor
+from langchain.chains import LLMChain
+from .local_files_utils import crawl_local_repo, extract_python_code_from_py, extract_python_code_from_ipynb
+from .templates import get_createproject_prompt_RAG_copy, prompt_data
+from .code_manipulation import prepare_code
+from .code_execution import execute_code
+from .chatlog import LangchainLoggingHandler
+
+# packages for memory
+from langchain.chains import create_history_aware_retriever, create_retrieval_chain
```

Ruff (0.11.9)

`langchain.chains.create_history_aware_retriever` imported but unused

Remove unused import

(F401)